### ISSUE TITLE
Port to 1.16.4?

### ISSUE THREAD

**Issue by ireece1** [MightyPirates/OpenComputers#3375]
I know it’s a very annoying, redundant question, but seeing as 1.16.4 is quickly shaping up to be the new “gold version” like 1.12.2 and 1.7.10 were, I was wondering if there’s any chance of OpenComputers coming to 1.16.4?

**Comment by DaDarkWizard** [comment]
It looks like it may be coming- looking at the build server, there’s two unstable 1.16 builds.

**Comment by Techokami** [comment]
Wot? I'm looking at the Jenkins now and don't see anything of the sort. Unless you're looking at a different build server, in which case can you please link?

**Comment by DaDarkWizard** [comment]
Yeah, just realized I can't read.. it was another mod.

**Comment by Creativious** [comment]
Yeah we really need this, though there is CC: Tweaked, but that doesn't have as much features as OpenComputers.

**Comment by ComputerErika** [comment]
Dear Creators

I know full well that updating mods to new versions is a pain, with every few versions having game-changing core stuff stoping mods from being updated easily. While to play modded I don't see this as one of my required mods, it is by far my favorite mod. Well above the rest, and it would pain me to see it die. If you don't plan on updating it, I'm sure there are people out there that would.

I just haven't seen or been able to find any news or such on this. While the mod hasn't been updated for over a year, the newest commit has been on Sep 8th. Which actually more than 2 months ago now that I realize... There are unofficial discords servers, but I can't find an official one. I just want to hear, if I should just give up hope on this, or if you plan on continuing support.

I know, I know, this isn't a review place, or a good way to message creators, but I don't know how else to do that without DMing them all personally, and while this is stupid probably really rude bothering them about this here, I feel like doing the DM would be even worse, and I feel out of options.

You're all still people, I know that, and I totally get if you don't want to continue this, it's a very complex mod, but that's why I love it so much more than the rest. It means so much to me and I've created so much on my own. It taught me Lua and got me back into programming. Hopefully, you can see how much this means to me. I hope you haven't had a bunch of people bothering you so much about this that it drives you leave it...

Last of, Sorry to OP for highjacking this "issue" request... Again, I just couldn't really think of a better place to post this, and didn't want to open a new request. I completly agree with you, and I hope 1.16 does become the new gold version of mods, I've missed a lot, and have seen them struggling to keep up.

From,
A Concered but Caring User,
ComputerEric

TL:DR
This is my favorite mod, I'd love to see it updated, but I understand if you can't. Sorry for bothering everyone.

**Comment by SquidDev** [comment]
> There are unofficial discords servers, but I can't find an official one. 

[Discord lives here](https://discord.gg/bYqKv7h).

**Comment by ComputerErika** [comment]
*facepalm* well I guess I am just very dumb

Thank you SquidDev

**Comment by Kristopher38** [comment]
I can only speak for myself, but I'm just gonna say to everyone wondering that the current maintainer is very busy with irl stuff, but very much still wants to support OC (at least that was the case a few months ago when he last spoke on discord).

**Comment by lucsoft** [comment]
Hey im working on a 1.16 Port! But sadly I'm the only dev on it.

**Comment by Scwestbrook** [comment]
> Hey im working on a 1.16 Port! But sadly I'm the only dev on it.

When do you think it will be available?
I'd help, but I have ZERO java knowledge, and only know HTML, and JS

**Comment by lucsoft** [comment]
no eta because i have a full time job and yea this takes some time
i have some tracking on my fork and you can see what im currently working on

**Comment by Elfinlocks** [comment]
Please note that Lucsoft's port is for FABRIC, not forge

**Comment by lucsoft** [comment]
yea it will have many benefits

**Comment by Scwestbrook** [comment]
OK, guess I'll have to find an alt. Have any suggestions?

**Comment by Elfinlocks** [comment]
Really the only other option is CC: Tweaked. It's the closest thing you'll find to this mod. Believe me, I'm sad to see it abandoned too.

**Comment by lucsoft** [comment]
Dude what are you guys talking about?

**Comment by Scwestbrook** [comment]
@Elfinlocks says the port will be fabric, not forge. Is it going to be forge?

**Comment by SirDavidLudwig** [comment]
I am also working on a 1.16 port, but for Forge. It's been a bit since I've worked on it as I've been finishing my degree but will be picking it back up in the next couple of weeks.

**Comment by Elfinlocks** [comment]
That's great news, David - I'm not much of a programmer, but if you need some assistance with QA or grunt coding, please hit me up on discord - Elfinlocks#1337. I'll do what I can to assist you to see this done.

**Comment by Scwestbrook** [comment]
@SirDavidLudwig, just so you know, I'm not amazing at coding or art either, but I'm really happy you're working on a Forge port. I'd help, but I can't really do much unless you need JS or Swift help.

**Comment by Scwestbrook** [comment]
Does anyone have any idea where I can learn Lua?

**Comment by MechanosG** [comment]
> 
> 
> Dude what are you guys talking about?

It sounds like they're trying to dismiss your project/comments after learning your port will be for Fabric instead of Forge.  Which seems to be a trend with (some) Forge supporters.  I for one, am thrilled to hear about a Fabric 1.16.4 port.  I wish you all the luck my dude, and I'll be following your project.  :+1:

**Comment by Elfinlocks** [comment]
> > Dude what are you guys talking about?
> 
> It sounds like they're trying to dismiss your project/comments after learning your port will be for Fabric instead of Forge. Which seems to be a trend with (some) Forge supporters. I for one, am thrilled to hear about a Fabric 1.16.4 port. I wish you all the luck my dude, and I'll be following your project. 👍

If you're only using OC on 1.16, I'm sure a fabric port is just dandy - some of us want the mod in actual modpacks.

**Comment by MechanosG** [comment]
> 
> 
> > > Dude what are you guys talking about?
> > 
> > 
> > It sounds like they're trying to dismiss your project/comments after learning your port will be for Fabric instead of Forge. Which seems to be a trend with (some) Forge supporters. I for one, am thrilled to hear about a Fabric 1.16.4 port. I wish you all the luck my dude, and I'll be following your project. +1
> 
> If you're only using OC on 1.16, I'm sure a fabric port is just dandy - some of us want the mod in actual modpacks.

Fabric has "actual modpacks".  It also loads faster, runs better, and has exclusive optimization mods to improve it even further.  Also, this - https://gist.github.com/jellysquid3/8b68b81a5e48462f8690284a0a3c89a1

Yes, Fabric is newer and some mods haven't gotten a Fabric version (and may never).  But many mods already have Fabric "alternatives", and due to a huge popularity spike this year, new mods are showing up every day.  Vehemently staying with Forge (dismissing any mention of fabric) just because you don't want to lose some mods you're used to..  well that's your decision I suppose.  But you're missing out.

**Comment by SquidDev** [comment]
Guys, you're maybe getting a little off topic here. This discussion is probably better placed on Discord or IRC.

**Comment by MechanosG** [comment]
The topic is a port for 1.16.4, and 2 people have mentioned ports in progress, one for Fabric, one for Forge.  The conversation passively went against the Fabric 1.16.4 port, so I commented in support of OC being on 1.16.4 Fabric, which is technically on topic.  Informing on why a Fabric port would be a good thing, was my only intention.  I apologize if it seemed like I was talking about non-OC mods too much, but the environment OC is used in is relevant I feel.

(Also, Discord/IRC is a sure fire way to be ignored, derailed, or banned by biased admins anyway.  I'm only interested in the topic of OC on 1.16.4, and feel both the Forge version and Fabric version should be celebrated.)

**Comment by ArsiTheFox** [comment]
> Does anyone have any idea where I can learn Lua?

CC: Tweaked a good way to learn :3, look up Gaming with Ben ComputerCraft challenge on YouTube. He taught me alot while also having fun with the mod <3

**Comment by ArsiTheFox** [comment]
@Scestbrook Also I can definitely teach you what I know
Discord: ArsiTheFoxxo#2533

**Comment by Creativious** [comment]
> no eta because i have a full time job and yea this takes some time
> i have some tracking on my fork and you can see what im currently working on

I'll check it out as I have experience with Java

**Comment by Creativious** [comment]
Also personally forge is better for heavy modification, while Fabric is better for light modification. Atleast in my testing.

**Comment by Creativious** [comment]
This deserves a port to both of these

**Comment by pivinx1** [comment]
Partially, Fabric has some heavy mods like VMComputers. Forge is more
suited for big modpacks because, well, there isn't much Fabric mods due to
the age of Fabric.
I would like to see OC on Fabric too.

czw., 31 gru 2020 o 06:36 Creativious <notifications@github.com> napisał(a):

**Comment by lucsoft** [comment]
It's not really true fabric has mixings and this allows you to modify stuff that doesn't have a api

But hey thanks @Creativious if you want to help feel free to do so

**Comment by Creativious** [comment]
> It's not really true fabric has mixings and this allows you to modify stuff that doesn't have a api
> 
> But hey thanks @Creativious if you want to help feel free to do so

I've never actually done anything with Fabric, only Forge and Spigot, but it won't take me long to learn it

**Comment by ireece1** [comment]
https://ci.cil.li/job/oc2-sedna/changes

“OpenComputers II”

OK NOW IM CURIOUS

**Comment by lucsoft** [comment]
Yes I teamed up with Sangar to bring OpenComputers 2 on Forge & Fabric

**Comment by Derpford** [comment]
re: big modpacks
having used both Fabric and Forge, I find that 1.16 Fabric performs *way* better than 1.16 Forge once you hit the 100+ mod mark
Forge has a number of prettier mods (I wish Create would get ported to Fabric...) but it suffers performance-wise, and Optifine is basically a must-have for many packs
in particular, Fabric's *startup times* are way faster even at the 100 mod mark, which not only makes it easier to play with, it also makes it easier as a packdev to test the pack

this is hard to measure accurately, unfortunately, just because you don't see a lot of crossplatform mod development (though there *was* a project for a little bit that was meant to give modders a library that they could use across both Fabric and Forge)
it's also not necessarily true across all the versions that Fabric and Forge are available for

finally, on a more personal note, I think Lex is an asshole--his attitude toward the community is awful, he thinks he's a god among men just because he inherited control of Forge, and the one piece of Forge that he's directly responsible for was the godawful model API change back in 1.10 that completely upended the entire modding community for the sake of using a model spec that wasn't nearly complete enough to work

In summary: basically the only reason I'm using Forge is because of established big-name mods, plus Create--and if OpenComputers goes to Fabric, that'll just be one more reason to stick to Fabric.

**Comment by Scwestbrook** [comment]
Exactly- I haven't migrated yet, just because I love Create. The 0.3 update is awesome!!

**Comment by ghost** [comment]
> Yes I teamed up with Sangar to bring OpenComputers 2 on Forge & Fabric

Any link to a github for this?

**Comment by 7eggert** [comment]
Crash with the current builds from https://ci.cil.li 
oc2-MC1.16-Forge
oc2-sedna-MC1.16-Forge

Since this is the only place for now, I'll upload the crash reports here:

[crash-2021-02-01_00.17.57-client.txt.gz](https://github.com/MightyPirates/OpenComputers/files/5901924/crash-2021-02-01_00.17.57-client.txt.gz)
[crash-2021-02-01_00.05.27-fml.txt.gz](https://github.com/MightyPirates/OpenComputers/files/5901925/crash-2021-02-01_00.05.27-fml.txt.gz)

**Comment by justinkluever** [comment]
> Crash with the current builds from https://ci.cil.li
> oc2-MC1.16-Forge
> oc2-sedna-MC1.16-Forge
> 
> Since this is the only place for now, I'll upload the crash reports here:
> 
> [crash-2021-02-01_00.17.57-client.txt.gz](https://github.com/MightyPirates/OpenComputers/files/5901924/crash-2021-02-01_00.17.57-client.txt.gz)
> [crash-2021-02-01_00.05.27-fml.txt.gz](https://github.com/MightyPirates/OpenComputers/files/5901925/crash-2021-02-01_00.05.27-fml.txt.gz)

in the fml crashlog it says that oc2-sedna (0.0.1 or higher) is not installed (it worked for me when i tested it)

**Comment by 7eggert** [comment]
Then I uploaded the one from before I tested it, too; I assumed them to belong together. Ignore the earlier one, please.

**Comment by lucsoft** [comment]
@7eggert Send crash reports please to the repo as an issue and not here so we can fix & track it better

**Comment by 7eggert** [comment]
Where is the repo for oc2?

**Comment by lucsoft** [comment]
well i have it pinned 

but yeah its here too https://github.com/fnuecke/oc2

**Comment by payonel** [comment]
oc2 is being worked on at noted here. oc has no plans to upgrade at this time
